src/LiveComponent/src/Attribute/AsLiveComponent.php

```diff
@@ -116,4 +124,11 @@ public static function liveListeners(object|string $component): array
 
         return $listeners;
     }
+
+    public static function getMethod(object|string $component): string
+    {
+        $attributes = (new \ReflectionClass($component))->getAttributes(self::class);
+
+        return $attributes[0]->newInstance()->serviceConfig()['method'] ?? 'post';
+    }
```

Comment:
I don't think this is needed. The other static methods are for things that need to read specific other attributes - like `LiveListener`. For an argument to `AsLiveComponent` (like `method`), when you need to read this, you can get it from the component metadata - example https://github.com/symfony/ux/blob/2.x/src/LiveComponent/src/Twig/LiveComponentRuntime.php#L48

Reply:
I've replaced the calls to `AsLiveComponent::getMethod` and removed this method.